ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,375 @@
+/* @flow weak */
+//normal imports
+import React from 'react';
+import _ from 'lodash';
+
+//other file imports
+import SetIntervalMixin from '../../helpers/set_interval_mixin.js';
+import {TextBody, TextFooter} from './texting_interface.jsx';
+import type {Response} from '../popup_question.jsx';
+
+//material-ui imports
+import FlatButton from 'material-ui/FlatButton';
+import Dialog from 'material-ui/Dialog';
+import Chip from 'material-ui/Chip';
+import Avatar from 'material-ui/Avatar';
+import {fullBlack} from 'material-ui/styles/colors';
+
+//material-ui icon imports
+import FaceIcon from 'material-ui/svg-icons/action/face';
+
+//
+const ONE_SECOND = 1000;
+
+export default React.createClass({
+  displayName: 'MobileInterface',
+
+  mixins: [SetIntervalMixin],
+  
+  getInitialState() {
+    var allExamples = [];
+    var messages = [];
+    const goodExamples = _.map(this.props.question.examples, (example) => { return {type: 'Good', text: example}; });
+    const badExamples = _.map(this.props.question.nonExamples, (example) => { return {type: 'Bad', text: example}; });
+    allExamples = _.shuffle(_.clone(goodExamples).concat(badExamples)); 
+    messages = this.getInitialMessages(this.props.question, this.props.scaffolding.shouldShowStudentCard);
+    return ({
+      messages,
+      allExamples,
+      selectedStudent: this.props.question.students[0],
+      initialResponse: undefined,
+      revisedResponse: undefined,
+      elapsedMs: 0,
+      exampleIndex: 0,
+      dialog: 'none',
+      messageId: 1
```

Comment:
This looks like an index into the `messages` array rather than a primary key that you'd use to look up a message, so i'd suggest `messageIndex` instead.